scene/gui/file_dialog.cpp

```diff
@@ -413,6 +413,10 @@ void FileDialog::update_file_name() {
 void FileDialog::update_file_list() {
 
 	tree->clear();
+
+	// Scroll back to the top after opening a directory
+	tree->get_vscroll_bar()->set_value(0);
```

Comment:
Is `update_file_list` only called when changing directories? How about when you would create a new directory?

Reply:
It seems that creating a directory auto-enters it, so at least this concerned is addressed. Let's merge and see how it goes.